plugin/spec-runner.vim

```diff
@@ -6,10 +6,35 @@ let s:COMMAND_FAILED = -1
 let s:NOT_IN_SPEC_FILE = -2
 let s:WARNING_UNABLE_TO_DETERMINE_RUNNER = 'Unable to determine correct spec runner'
 
+let g:spec_runner_available_runners = {
+      \ 'rspec' : 'rspec',
+      \ 'teaspoon' : { 'no_preloader': 'teaspoon', 'zeus': 'rake teaspoon' },
+      \ }
+
 if ! exists('g:spec_runner_dispatcher')
   let g:spec_runner_dispatcher = '!echo "{command}" && {command}'
 endif
 
+function! g:SpecRunner_detect_teaspoon()
+  return s:InJavascriptFile() && s:InGemfile('teaspoon')
+endfunction
+
+function! g:SpecRunner_detect_rspec()
+  return s:InRspecFile()
+endfunction
+
+function! s:DetectRunnerName()
+  for runner_name in keys(g:spec_runner_available_runners)
+    let function_name = 'g:SpecRunner_detect_' . runner_name
+    execute 'let worked = ' . function_name . '()'
```

Comment:
This is gross, but `let worked = execute(function_name . "())` doesn't work. I'd welcome a better way.